packages/dev/core/src/Materials/meshDebugPluginMaterial.ts

```diff
@@ -0,0 +1,442 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { MaterialPluginBase } from "./materialPluginBase";
+import type { Scene } from "../scene";
+import type { UniformBuffer } from "./uniformBuffer";
+import type { Nullable } from "../types";
+import { MaterialDefines } from "./materialDefines";
+import type { PBRBaseMaterial } from "./PBR/pbrBaseMaterial";
+import type { StandardMaterial } from "./standardMaterial";
+import { RegisterClass } from "../Misc/typeStore";
+import { Color3, Vector3 } from "core/Maths/math";
+import type { Mesh } from "core/Meshes/mesh";
+import { Logger } from "core/Misc/logger";
+import { expandToProperty, serialize } from "core/Misc/decorators";
+import type { AbstractMesh } from "core/Meshes/abstractMesh";
+
+/**
+ * Supported visualizations of MeshDebugPluginMaterial
+ */
+export enum MeshDebugMode {
+    /**
+     * Material without any mesh debug visualization
+     */
+    NONE = -1,
+    /**
+     * A wireframe of the mesh
+     * NOTE: For this mode to work correctly, bar() must first be called on mesh.
+     */
+    TRIANGLES = 0,
+    /**
+     * A wireframe of the mesh, with dots drawn over vertices
+     * NOTE: For this mode to work correctly, foo() must first be called on mesh.

```

Comment:
Not sure about the note here ?

Reply:
Huh. I could've sworn I changed these notes to be "... tripleTriangles() must first be called..." Whoops.

These are notes for proper usage of the mode, since VERTICES wouldn't work without first tripling the triangles rendered.